packages/frontend/src/components/Election/Voting/VotePage.tsx

```diff
@@ -97,6 +102,7 @@ const VotePage = () => {
     setPages(newPages)
   }
   const submit = async () => {
+    socket.emit('new_vote');
```

Comment:
I think the triggering of this event should be done from the backend at the end of the post request. If you export io from the socket handler you can emit directly from it anywhere in the server.

For example, I took one of their examples of a [chatroom ](https://codesandbox.io/p/devbox/github/socketio/chat-example/tree/cjs/step5?file=%2Findex.js%3A23%2C1) and appended this at the bottom of the server
```
setInterval(function () {
  io.emit('chat message', 'hello');
}, 1000);
```
And it sends a message to the client every second without the client having to trigger the event.

Reply:
Updated!

I wish I could find a way to avoid typing io as "Server | null", but let me know if you have ideas.

I also initially tried having a ``emit('new_vote')`` in the castVoteController and a separate ``on('new_vote')`` in the sockethandler, but that didn't work.

Reply:
the null typing is probably due to the order things are initialized. The express app and all the routes/controllers are created first in "app" and then that is used to create io. So at the time the castVoteController imports it io doesn't exist yet.